app/models/bot/alegre.rb

```diff
@@ -119,49 +114,47 @@ def self.merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, pm)
     end
   end
 
-  def self.get_threshold_for_video_query(pm, automatic=false)
-    key = 'video_similarity_threshold'
-    key = "automatic_#{key}" if automatic
-    return {value: CheckConfig.get(key).to_f, key: key, automatic: automatic}
-  end
-
-  def self.get_threshold_for_image_query(pm, automatic=false)
-    key = 'image_similarity_threshold'
-    key = "automatic_#{key}" if automatic
-    return {value: CheckConfig.get(key).to_f, key: key, automatic: automatic}
-  end
-
-  def self.get_threshold_for_text_query(pm, automatic=false)
-    model = self.matching_model_to_use(pm)
-    key = "text_similarity_threshold"
+  def self.get_threshold_for_query(type, pm, automatic = false)
+    key = "#{type}_similarity_threshold"
     key = "automatic_#{key}" if automatic
-    key = "vector_#{key}" if model != Bot::Alegre::ELASTICSEARCH_MODEL
-    return {value: CheckConfig.get(key).to_f, key: key, automatic: automatic}
+    if type == 'text' && !pm.nil?
+      model = self.matching_model_to_use(pm)
+      key = "vector_#{key}" if model != Bot::Alegre::ELASTICSEARCH_MODEL
+    end
+    { value: CheckConfig.get(key).to_f, key: key, automatic: automatic }
   end
 
   def self.get_similar_items(pm)
+    type = nil
     if pm.is_text?
-      suggested_or_confirmed = self.get_merged_items_with_similar_text(pm, self.get_threshold_for_text_query(pm))
-      confirmed = self.get_merged_items_with_similar_text(pm, self.get_threshold_for_text_query(pm, true))
-      self.merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, pm)
+      type = 'text'
     elsif pm.is_image?
-      suggested_or_confirmed = self.get_items_with_similar_image(pm, self.get_threshold_for_image_query(pm))
-      confirmed = self.get_items_with_similar_image(pm, self.get_threshold_for_image_query(pm, true))
-      self.merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, pm)
+      type = 'image'
     elsif pm.is_video?
-      suggested_or_confirmed = self.get_items_with_similar_video(pm, self.get_threshold_for_video_query(pm))
-      confirmed = self.get_items_with_similar_video(pm, self.get_threshold_for_video_query(pm, true))
+      type = 'video'
+    end
+    unless type.blank?
+      suggested_or_confirmed = self.get_items_with_similarity(type, pm, self.get_threshold_for_query(type, pm))
+      confirmed = self.get_items_with_similarity(type, pm, self.get_threshold_for_query(type, pm, true))
       self.merge_suggested_and_confirmed(suggested_or_confirmed, confirmed, pm)
     else
       {}
     end
   end
 
+  def self.get_items_with_similarity(type, pm, threshold)
+    if type == 'text'
+      self.get_merged_items_with_similar_text(pm, threshold)
```

Comment:
I think we should probably also `reject_same_case` on these as well, no @caiosba ?

Reply:
What you mean, @DGaffney ? Sorry, I didn't understand...

Reply:
me to didn't understand,  I just group methods to overcome CC and still using same existing code

Reply:
We have a function `reject_same_case` that removes from the list of similar matches the ProjectMedia we are currently matching to other cases (e.g. don't return PM ID of 1 as a match for PM ID of 1). We use it on line 149 here, but not on these text cases - does that help clarify?

Reply:
Ah, thanks... @melsawy please confirm we're keeping that behavior.

Reply:
I see that we already using the behavior for `reject_sama_case` for text cases https://github.com/meedan/check-api/blob/feature/CHECK-675-add-too-short-text-as-suggestions/app/models/bot/alegre.rb#L410